src/main/java/org/minions/devfund/alejandro/Squarelotron.java

```diff
@@ -0,0 +1,179 @@
+package org.minions.devfund.alejandro;
+
+/**
+ * Squareotron class.
+ */
+public class Squarelotron {
+
+    private int[][] squarelotronMatrix;
+    private int size;
+
+    /**
+     * It fills the 2-dimensional array with the numbers 1 to n squared, in order.
+     * It also sets the size instance variable to be n.
+     * @param n The size for the matrix
+     */
+    public Squarelotron(int n) {
+        this.size = n;
+        n = 1;
+        squarelotronMatrix = new int[this.size][this.size];
+        for (int x = 0; x < this.size; x++) {
+            for (int y = 0; y < this.size; y++) {
+                squarelotronMatrix[x][y] = n++;
+            }
+        }
+    }
+
+    /**
+     * This method performs the Upside-Down Flip of the squarelotronMatrix,
+     * and returns the new squarelotronMatrix. The original squarelotronMatrix should not be modified.
+     * @param ring the ring that will be flipped.
+     * @return A modified instance of a new Squarelotron.
+     */
+    public Squarelotron upsideDownFlip(int ring) {
+
+        Squarelotron resultSquare = new Squarelotron(this.size);
+        int[][] swappedMatrix = this.getSquarelotronMatrix();
+
+        if (ring > 0 && ring <= this.size / 2) {
+
+            swapMatrix(swappedMatrix);
+            resultSquare.squarelotronMatrix =
+                    flipSquarelotron(ring, resultSquare.getSquarelotronMatrix(), swappedMatrix);
+        }
+        return resultSquare;
+    }
+
+    /**
+     * This method will swap a matrix.
+     * @param matrix The matrix that will be swapped.
+     * @return the same matrix but swapped.
+     */
+    private int[][] swapMatrix(final int[][] matrix) {
+
+        int controlSize = this.size - 1;
+        int[] row;
+        int[] row1;
+
+        for (int i = 0; i < this.size / 2; i++) {
+            row = matrix[i];
+            row1 = matrix[controlSize];
+            matrix[i] = row1;
+            matrix[controlSize] = row;
+            controlSize--;
+        }
+        return matrix;
+    }
+
+    /**
+     * This method will flip a matrix in a determined ring.
+     * @param ring the ring that will be flipped in the square.
+     * @param squarelotronFlipedMatrix is the matrix that will be flipped.
+     * @param swapedMatrix the swapped matrix.
+     * @return the flipped matrix.
+     */
+    private int[][] flipSquarelotron(int ring, final int[][] squarelotronFlipedMatrix, final int[][] swapedMatrix) {
+
+        for (int i = ring - 1; i <= this.size - ring; i++) {
+            for (int j = ring - 1; j <= this.size - ring; j++) {
+                if (i == ring - 1 || j == ring - 1 || i == this.size - ring || j == this.size - ring) {
```

Comment:
This if condition is repeated in line 99
Can you add a private method called isPositionInRing() ?